(#5355) - Change sprites when someone is buckled into the rollerbed

HuntorWorks opened the pull request:
CL: [New] When a player is buckled into the roller bed, the sprite now changes based on status of the bed.  When the player is buckled the bed will raise, when the bed is free, the bed will lower. 

CL: [Improvement] Created event system on the back end which developers can now use should any more features need to be added when someone is buckled / unbuckled from the bed.

HuntorWorks committed:
#5355 - Added sprites to work when someone is buckled into the roller bed

Added sprite script to roller bed prefab, and created OnBuckle and OnUnbuckle events in BuckleInteract.cs to subscribe for when someone leaves the roller bed and when someone is buckled to change the sprite, depending on bed status.

MaxIsJoe commented on UnityProject/Assets/Scripts/Items/Medical/RollerBedSpriteChange.cs:
```diff
@@ -0,0 +1,38 @@
+using UnityEngine;
+using Objects;
+
+public class RollerBedSpriteChange : MonoBehaviour
```
This script needs to be namespaced

HuntorWorks replied:
name spaced to what?

HuntorWorks replied:
Changed to namespace Items.Medical

MaxIsJoe commented on UnityProject/Assets/Scripts/Items/Medical/RollerBedSpriteChange.cs:
```diff
@@ -0,0 +1,38 @@
+
+public class RollerBedSpriteChange : MonoBehaviour
+{
+    ///<summary>
```
Wrong placement of summery, put it ontop of the class not the `BuckleInteract` variable

MaxIsJoe commented on UnityProject/Assets/Scripts/Objects/Furniture/BuckleInteract.cs:
```diff
@@ -17,6 +18,12 @@ public class BuckleInteract : MonoBehaviour, ICheckedInteractable<MouseDrop>, IC
 		private Integrity integrity;
 
+		///<summary>
+		/// Events created to change the sprite of the roller bed
```
Events are a bit universal in their use cases, so a better description would be:
"Event that get invoked when a player Buckles into an object that has this component"

and create two summaries for both events so the IDE picks them up.
"Event that get invoked when a player Unbuckles into an object that has this component"

MaxIsJoe commented on UnityProject/Assets/Scripts/Objects/Furniture/BuckleInteract.cs:
```diff
@@ -133,6 +140,7 @@ public void BucklePlayer(PlayerScript playerScript)
 			}
 
 			objectPhysics.BuckleObjectToThis(playerScript.playerMove);
+			OnBuckleEvent?.Invoke();
```
This is a nitpick, but you can ignore this one if you'd like; Events are best invoked last when all the logic above it has passed first. In this scenario there _shouldn't_ be an issue per say, but since things are being done in `occupiedSpriteHandler` this might override other stuff the events are trying to do in the future.

HuntorWorks committed:
PR Merge Request

Added and reworded summary for the events,  namespaced the rollerbed script and moved Event calls to the last line of the code.

HuntorWorks committed:
Removed Unused using statements

Removed unused using statements to satisfy codacy errors.

HuntorWorks committed:
Removed Unused Using statements

To satisfy codacy
